sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-response/_tests_/ece-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,662 @@
+        process.env.INSTITUTION_RESPONSE_FOLDER,
+        locationSKIP.institutionCode,
+        ECE_RESPONSE_FILE_NAME,
+      );
+
+      // Queued job.
+      const job = createMock<Job<void>>();
+
+      // Modify the data in mock file to have invalid application number.
+      // Note: The disbursement id is already an invalid value in file
+      // due to the placeholder DISBNUMBER.
+      mockDownloadFiles(
+        sftpClientMock,
+        [ECE_RESPONSE_FILE_NAME],
+        (fileContent: string) => {
+          return fileContent.replace("APPLNUMBER", "          ");
+        },
+      );
+
+      // Act
+      const processResult = await processor.processECEResponse(job);
+
+      // Assert
+      const expectedResult: ProcessSummaryResult = new ProcessSummaryResult();
+      expectedResult.summary = [
+        `Starting download of file ${confirmEnrolmentResponseFile}.`,
+        `The file ${confirmEnrolmentResponseFile} has been deleted after processing.`,
+      ];
+      expectedResult.errors = [
+        "Invalid unique index number for the disbursement record, Invalid application number at line 2.",
+        `Error processing the file ${confirmEnrolmentResponseFile}. Error: The file consists invalid data and cannot be processed.`,
+      ];
+      expect(processResult).toStrictEqual([expectedResult]);
+      // Expect the delete method to be called.
+      expect(sftpClientMock.delete).toHaveBeenCalled();
+    });
+  },
+);
+
+/**
+ * Create institution locations to be used for testing.
+ * @param e2eDataSources e2e data sources.
+ * @returns institution locations.
+ */
+async function createInstitutionLocations(
+  e2eDataSources: E2EDataSources,
+): Promise<{
+  institutionLocationCONF: InstitutionLocation;
+  institutionLocationDECL: InstitutionLocation;
+  institutionLocationSKIP: InstitutionLocation;
+  institutionLocationFAIL: InstitutionLocation;
+}> {
+  const institution = await e2eDataSources.institution.save(
+    createFakeInstitution(),
+  );
+  const institutionLocationCONF = await findOrCreateInstitutionLocation(
+    institution,
+    "CONF",
+    e2eDataSources,
+  );
+  const institutionLocationDECL = await findOrCreateInstitutionLocation(
+    institution,
+    "DECL",
+    e2eDataSources,
+  );
+
+  const institutionLocationSKIP = await findOrCreateInstitutionLocation(
+    institution,
+    "SKIP",
+    e2eDataSources,
+  );
+
+  const institutionLocationFAIL = await findOrCreateInstitutionLocation(
+    institution,
+    "FAIL",
+    e2eDataSources,
+  );
+
+  return {
+    institutionLocationCONF,
+    institutionLocationDECL,
+    institutionLocationSKIP,
+    institutionLocationFAIL,
+  };
+}
+
+/**
+ * Look for an existing institution location by institution code
+ * if not found create one.
+ * @param institution institution.
+ * @param institutionCode institution code.
+ * @param e2eDataSources e2e data sources.
+ * @returns institution location.
+ */
+async function findOrCreateInstitutionLocation(
```

Comment:
Since these methods are specific for these tests I do not see a problem with leaving them here, but when we have a good amount I would always prefer to create them on a separated file. We did it already for other E2E tests also.